src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -640,6 +640,55 @@ public override async Task<IReadOnlyList<BulkStateItem>> GetBulkStateAsync(strin
             return bulkResponse;
         }
 
+        /// <inheritdoc/> 
+        public override async Task<IReadOnlyList<BulkStateItem<TValue>>> GetBulkStateAsync<TValue>(string storeName,
+            IReadOnlyList<string> keys, int? parallelism, IReadOnlyDictionary<string, string> metadata = default,
+            CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(storeName, nameof(storeName));
```

Comment:
Would it make sense to consolidate the setup/execution of the bulk operation, as the only real difference is in handling of the response, in order to ensure these methods don't ever vary?

Reply:
Are you ok if I rewrite both methods to use a private shared method? I took this approach to avoid deserializing the value from the JSON byte string twice and to avoid impacting the original method, but I agree - for forward maintainability, this could be improved. I'd propose a shared private method that returns the Key, ETag and ByteString values and leave it to either public method to then deserialize once in the manner most appropriate to either.

Reply:
Updated to use the shared private method so each of the public-facing methods perform the deserialization against the ByteString only once.